lib/sshkit/ssh/channel.ex

```diff
@@ -39,6 +39,31 @@ defmodule SSHKit.SSH.Channel do
     end
   end
 
+  @doc """
+  Activates a subsystem on a channel.
+
+  On success, returns `:ok`. On failure, returns `{:error, reason}`.
+
+  For more details, see [`:ssh_connection.subsystem/4`](http://erlang.org/doc/man/ssh_connection.html#subsystem-4).
+  """
+  @spec subsystem(channel :: struct(), subsystem :: String.t(), options :: list()) ::
```

Comment:
👍 for the doctests

Could you please also add some _"normal"_ unit tests for this function? It could cover some more edgecases and give a bit more insight into the underlying behavior.